lib/coffee-provider.coffee

```diff
@@ -1,20 +1,29 @@
-coffee = null
+coffee = CSON = null
 
 module.exports =
 class CoffeeProvider
   fromScopeName: 'source.coffee'
   toScopeName: 'source.js'
+  indent: 2
 
   transform: (code, {sourceMap} = {}) ->
-    coffee ?= require 'coffee-script'
-    options =
-      sourceMap: sourceMap ? false
-      bare: atom.config.get('source-preview.coffeeProviderOptionBare')
+    if atom.workspace.getActiveTextEditor().getTitle().endsWith '.cson'
+      #@toScopeName = 'source.json'
```

Comment:
Ideally this would also set the grammar of the preview pane to `source.json`, and when I enable this (and line 16) it does work. However it causes a kind of 'delay', meaning if I preview a .cson file, the grammar of the preview pane will be correctly set to JSON (`source.json`) but if I close that and then switch over to a CoffeeScript (.coffee) file and preview _that_ the grammar of the preview pane stays as `source.json`. But if I close _that_ and then re-preview it, then the grammar is correctly set to `source.js`. This also happens in the reverse order…

I'm not quite sure how to address this… any ideas?

For now, just leaving the grammar of the preview pane as JavaScript (`source.js`) actually doesn't make a difference to the syntax highlighting in this case, so this isn't a big deal… it just doesn't set the _proper_ grammar.